File: code/checking-type.kt
var movieCount = 0
var songCount = 0

class Movie
class Song

fun main10() {
    val library: List<Any> = listOf("first")
    for (item in library) {
        if (item is Movie) {
            ++movieCount
        } else if (item is Song) {
            ++songCount
        }
    }
}

File: code/checking-type.ts
let movieCount = 0;
let songCount = 0;

class Movie {
}

class Song {
}

let library: unknown[] = [new Movie(), new Song()]

for (const item of library) {
    if (item instanceof Movie) {
        ++movieCount;
    } else if (item instanceof Song) {
        ++songCount;
    }
}
